x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/components/no_data_content.tsx

```diff
@@ -0,0 +1,128 @@
+        <FormattedMessage
+          id="xpack.infra.metricsTable.loadingContentTitle"
+          defaultMessage="Loading metrics"
+        />
+      </h2>
+    }
+  />
+);
+
+export const MetricsTableNoIndicesContent = () => {
+  const integrationsLinkProps = useLinkProps({ app: 'integrations', pathname: 'browse' });
+
+  return (
+    <EuiEmptyPrompt
+      data-test-subj="metricsTableLoadingContent"
+      iconType="logoMetrics"
+      title={<h2>{noMetricIndicesPromptTitle}</h2>}
+      body={<p>{noMetricIndicesPromptDescription}</p>}
+      actions={
+        <EuiButton color="primary" fill {...integrationsLinkProps}>
+          {noMetricIndicesPromptPrimaryActionTitle}
+        </EuiButton>
+      }
+    />
+  );
+};
+
+export const MetricsTableEmptyIndicesContent = () => {
+  return (
+    <EuiEmptyPrompt
+      body={
+        <EuiDescriptionList compressed>
+          <EuiDescriptionListTitle>
+            <FormattedMessage
+              id="xpack.infra.metricsTable.emptyIndicesPromptTimeRangeHintTitle"
+              defaultMessage="Expand your time range"
+            />
+          </EuiDescriptionListTitle>
+          <EuiDescriptionListDescription>
+            <FormattedMessage
+              id="xpack.infra.metricsTable.emptyIndicesPromptTimeRangeHintDescription"
+              defaultMessage="Try searching over a longer period of time."
+            />
+          </EuiDescriptionListDescription>
+          <EuiDescriptionListTitle>
+            <FormattedMessage
+              id="xpack.infra.metricsTable.emptyIndicesPromptQueryHintTitle"
+              defaultMessage="Adjust your query"
+            />
+          </EuiDescriptionListTitle>
+          <EuiDescriptionListDescription>
+            <FormattedMessage
+              id="xpack.infra.metricsTable.emptyIndicesPromptQueryHintDescription"
+              defaultMessage="Try searching for a different combination of terms."
+            />
+          </EuiDescriptionListDescription>
+        </EuiDescriptionList>
+      }
+      color="subdued"
+      data-test-subj="metricsTableEmptyIndicesContent"
+      icon={<NoResultsIllustration />}
+      layout="horizontal"
+      title={
+        <h2>
+          <FormattedMessage
+            id="xpack.infra.metricsTable.emptyIndicesPromptTitle"
+            defaultMessage="No results match your search criteria"
+          />
+        </h2>
+      }
+      titleSize="m"
+    />
+  );
+};
+
+const NoResultsIllustration = () => {
+  const { colorMode } = useEuiTheme();
+
+  const illustration =
+    colorMode === COLOR_MODES_STANDARD.dark
+      ? noResultsIllustrationDark
+      : noResultsIllustrationLight;
+
+  return (
+    <EuiImage alt={noResultsIllustrationAlternativeText} size="fullWidth" src={illustration} />
+  );
+};
+
+const noResultsIllustrationAlternativeText = i18n.translate(
+  'xpack.infra.metricsTable.noResultsIllustrationAlternativeText',
+  { defaultMessage: 'A magnifying glass with an exclamation mark' }
+);
```

Comment:
Is there a benefit to extracting this? I suppose this only pulls the translation once, should we do this in more places?

Reply:
I prefer this when we don't use `<FormattedMessage`> because
- it makes the JSX less noisy
- improves performance a bit
- allows for using the same string in multiple places without performing the translation multiple times

But so far it didn't cross my threshold of "we need a rule for this". :shrug: